instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py

```diff
@@ -299,6 +301,77 @@ def _inject_propagation_headers(headers, args, kwargs):
         kwargs["headers"] = _headers.raw
 
 
+def _extract_response(
+    response: typing.Union[
+        httpx.Response, typing.Tuple[int, Headers, httpx.SyncByteStream, dict]
+    ]
+) -> typing.Tuple[int, Headers, httpx.SyncByteStream, dict, str]:
+    if isinstance(response, httpx.Response):
+        status_code = response.status_code
+        headers = response.headers
+        stream = response.stream
+        extensions = response.extensions
+        http_version = response.http_version
+    else:
+        status_code, headers, stream, extensions = response
+        http_version = extensions.get("http_version", b"HTTP/1.1").decode(
+            "ascii", errors="ignore"
+        )
+
+    return (status_code, headers, stream, extensions, http_version)
+
+
+def _apply_request_client_attributes_to_span(
+    span_attributes, url, method_original, span_name, semconv
+):
+    # http semconv transition: http.method -> http.request.method
+    _set_http_method(span_attributes, method_original, span_name, semconv)
+    # http semconv transition: http.url -> url.full
+    _set_http_url(span_attributes, str(url), semconv)
+    try:
```

Comment:
Is the try except there for in the case `url` does not have `host` or `port` fields?

Reply:
Yes

Reply:
In that case, `host` and `port` are properties of httpx.URL and they have defaults. Think I can remove this.

Reply:
I don't believe we have exception handling in most places for expected fields from using instrumented libraries. At best I think we can check if the fields are there with `hasattr` and if they do not exist, we simply exclude the span attribute.